packages/adminrouter/extra/src/test-harness/modules/generic_test_code/generalised_tests.py

```diff
@@ -443,12 +497,16 @@ def test_if_upstream_headers_are_correct(
             path,
             jwt_forwarded_test,
             skip_authcookie_filtering_test,
-            ):
+            vhost):
+
+        req_headers = copy.deepcopy(valid_user_header)
+        if vhost is not None:
+            req_headers["Host"] = vhost
```

Comment:
I don't fully understand this code by do we append here (and one other place) and override the headers a few other places?

Reply:
Only append, it should be consistent now (see the comment above).